Framework/DataHandling/src/LoadNGEM.cpp

```diff
@@ -44,6 +44,85 @@ uint64_t swapUint64(uint64_t word) {
   return (word << 32) | (word >> 32);
 }
+
+/**
+ * @brief Correct a big endian event to be compatible with x64 and x86
+ * architecture.
+ *
+ * @param bigEndian The big endian formatted event.
+ * @param smallEndian The resulting small endian formatted event.
+ */
+void correctForBigEndian(EventUnion *&bigEndian, EventUnion &smallEndian) {
+  smallEndian.splitWord.words[0] = swapUint64(bigEndian->splitWord.words[1]);
+  smallEndian.splitWord.words[1] = swapUint64(bigEndian->splitWord.words[0]);
+}
+
+/**
+ * @brief Add a frame to the main set of histograms.
+ *
+ * @param rawFrames The number of T0 Events detected so far.
+ * @param goodFrames The number of good frames detected so far.
+ * @param eventCountInFrame The number of events in the current frame.
+ * @param minEventsReq The number of events required to be a good frame.
+ * @param maxEventsReq The max events allowed to be a good frame.
+ * @param frameEventCounts A vector of the number of events in each good frame.
+ * @param histograms The main set of histograms for the data so far.
+ * @param histogramsInFrame The set of histograms for the current frame.
+ */
+void addFrameToOutputWorkspace(
+    int &rawFrames, int &goodFrames, const int &eventCountInFrame,
+    const int &minEventsReq, const int &maxEventsReq,
+    MantidVec &frameEventCounts,
+    std::vector<DataObjects::EventList> &histograms,
+    std::vector<DataObjects::EventList> &histogramsInFrame) {
+  ++rawFrames;
+  if (eventCountInFrame >= minEventsReq && eventCountInFrame <= maxEventsReq) {
+    // Add number of event counts to workspace.
+    frameEventCounts.emplace_back(eventCountInFrame);
+    ++goodFrames;
+
+    PARALLEL_FOR_NO_WSP_CHECK()
+    // Add histograms that match parameters to workspace
+    for (auto i = 0; i < NUM_OF_SPECTRA; ++i) {
+      if (histogramsInFrame[i].getNumberEvents() > 0) {
+        histograms[i] += histogramsInFrame[i];
+        histogramsInFrame[i].clear();
+      }
+    }
+  }
+}
+
+/**
+ * @brief Creates an event workspace and fills it with the data.
+ *
+ * @param maxToF The largest ToF seen so far.
+ * @param binWidth The width of each bin.
+ * @param histograms The main histogram event data.
+ * @param dataWorkspace The workspace to add the data to.
+ */
+void createEventWorkspace(const int &maxToF, const double &binWidth,
+                          std::vector<DataObjects::EventList> &histograms,
+                          DataObjects::EventWorkspace_sptr &dataWorkspace) {
+  std::vector<double> xAxis;
+  // Round up number of bins needed and reserve the space in the vector.
+  xAxis.reserve(int(std::ceil(maxToF / binWidth)));
+  for (auto i = 0; i < (maxToF / binWidth); i++) {
+    xAxis.push_back(i * binWidth);
+  }
+
+  dataWorkspace = DataObjects::create<DataObjects::EventWorkspace>(
+      NUM_OF_SPECTRA, HistogramData::Histogram(HistogramData::BinEdges(xAxis)));
+  PARALLEL_FOR_NO_WSP_CHECK()
+  for (auto spectrumNo = 0u; spectrumNo < histograms.size(); ++spectrumNo) {
```

Comment:
Can `spectrumNo` be renamed, this is not a spectrum number and just an index.

In fact on first glance calling it spectrum number made the loop look unsafe as I assumed `getSpectrum` actually did accept spectrum numbers.